doc/file-upload.md

```diff
@@ -0,0 +1,293 @@
+# Uploading files from an HTML form
+
+When uploading a file from an HTML form, the file content is not sent as the request body.
+This is because an HTML form may contain more than one input, and in a POST request all input values will be sent in the request body.
+
+A form which has an `<input type="file">` must use `enctype="multipart/form-data"`, according to the [specification](https://www.ietf.org/rfc/rfc1867.txt).
+
+The file content can be sent as raw data, or base64 encoded, according to the [OpenAPI specification](https://swagger.io/docs/specification/describing-request-body/file-upload/):
+"Files use a `type: string` schema with `format: binary` or `format: base64`, depending on how the file contents will be encoded."
+
+How this all works when using Roaster is illustrated in the examples below.
+
+## Uploading a single file
+
+In this example, the filename is sent as a separate parameter.
+This is not strictly necessary, but it shows how to send a filename which may be different from the original filename as part of the URL path.
+
+First look at the HTML form.
+
+```html
+    <form id="singleFileUploadForm"
+        action="#"
+        method="POST" 
+        enctype="multipart/form-data"
+        onsubmit="uploadSingleFile"
+    >
+        <input type="file" name="file" id="singleFileUploadInput"/><br/>
+        <input type="submit" value="Upload text file"/>
+    </form>
+```
+
+The javascript function used for upload:
```

Comment:
would be great to add example not requiring javascript so just a plain html form

Reply:
The one with multi file uploads is working without any JS. But this example reads the filename and modifies the action.